wremnants/CardTool.py

```diff
@@ -854,13 +870,24 @@ def fillCardWithSyst(self, syst):
         systnamesPruned = [s for s in systNames if not self.isExcludedNuisance(s)]
         systNames = systnamesPruned[:]
         for chan in self.channels:
-            for systname in systNames:
-                shape = "shape" if not systInfo["noConstraint"] else "shapeNoConstraint"
-                # do not write systs which should only apply to other charge, to simplify card
-                self.cardContent[chan] += f"{systname.ljust(self.spacing)} {shape.ljust(self.systTypeSpacing)} {''.join(include_chan[chan])}\n"
+            systNamesChan = [x.replace("CHANNEL",chan) for x in systNames]
+            for systname in systNamesChan:
+                systShape = shape
+                include_line = include_chan[chan]
+                # some customization based on nuisance
+                if systInfo["customizeNuisance"]:
```

Comment:
What customisation is happening here...?

Reply:
So currently this works to modify the "scale" parameter in the data card (e.g. for alpha we have 0.67), and the type of nuisance (shape vs shapeNoConstraint or potentially something else).
In both cases, this overrides the global settings for the nuisances matching the regular expression used to key the dictionary, the not matching parameters will still keep the original global setting.

Reply:
Can you just allow the type of scale or noi to determine this instead? like if it's a dictionary instead of a float/bool

Reply:
I don't really understand the function for "customizeNuisance". 
For setting shape vs shapeNoConstraint we already have the "noConstraint" key, and for setting the scale you have implemented the "scale" key. I don't think we need to keep track of the initial and changed settings via the "customizeNuisance", if I understand correctly that this is its purpose. 
If the card setting have to be changed I think they can be changed directly. Ideally, they are set to the desired value in the initialization

Reply:
This was needed because I wanted to treat some nuisances in the same histogram differently, using the default scale and shape type for some and different ones for some (possibly also with multiple configurations). 

Initially this was being done by modifying the scale and noConstrain flags directly to be a dictionary themselves (so if they are float it is interpreted as a common value, otherwise it is configured based on nuisances). This would have been more flexible, but the internal implementation was also a little more convoluted, so I thought that having a new argument for special configurations was simpler and better (also because this is mainly for tests). For the record, there was a time when I had implemented the possibility for scale to be a dictionary, but since there was not a use case (the fake syst on which it was used had been removed) I was told I should remove it.

Reply:
In any case, I was mentioning alpha to clarify what the meaning of the scale was, but I am not using that feature for the alpha nuisance